tests/sentry/data_export/test_tasks.py

```diff
@@ -286,3 +358,23 @@ def test_discover_snuba_error(self, emailer, mock_query):
             assemble_download(de.id)
         error = emailer.call_args[1]["message"]
         assert error == "Internal error. Your query failed to run."
+
+    @patch("sentry.data_export.models.ExportedData.finalize_upload")
+    @patch("sentry.data_export.models.ExportedData.email_failure")
+    def test_discover_integrity_error(self, emailer, finalize_upload):
+        de = ExportedData.objects.create(
+            user=self.user,
+            organization=self.org,
+            query_type=ExportQueryType.DISCOVER,
+            query_info={"project": [self.project.id], "field": ["title"], "query": ""},
+        )
+        finalize_upload.side_effect = IntegrityError("test")
+        with self.tasks():
+            assemble_download(de.id)
+        error = emailer.call_args[1]["message"]
+        assert error == "Failed to save the assembled file."
+
+
+class MergeExportBlobsTest(TestCase, SnubaTestCase):
+    def test_task_persistent_name(self):
+        assert merge_export_blobs.name == "sentry.data_export.tasks.merge_blobs"
```

Comment:
Should you have tests that check that the ExportData.blob objects end up on the file that instance that is created and the success email is sent?

Reply:
There's 2 tests, `test_issue_by_tag` and `test_discover` that already exists which asserts that the file is created with the correct contents. I can add in an assertion that the success email is sent.